config/config.go

```diff
@@ -545,6 +545,18 @@ func ParseConfigFile(filename string, terragruntOptions *options.TerragruntOptio
 		return nil, err
 	}
 
+	// Initial evaluation of configuration to load flags like IamRole which will be used for final parsing
+	// https://github.com/gruntwork-io/terragrunt/issues/667
+	if include == nil && terragruntOptions.IamRole == "" {
+		iamConfig, err := PartialParseConfigString(configString, terragruntOptions, include, filename, []PartialDecodeSectionType{TerragruntFlags})
+		if err != nil {
+			return nil, err
+		}
+		if iamConfig.IamRole != "" {
+			terragruntOptions.IamRole = iamConfig.IamRole
+		}
+	}
```

Comment:
We probably want this logic in `ParseConfigString`, since there are more entrypoints to that function than `ParseConfigFile`.

Reply:
Also, put this in a separate function to make it easier to extend in the future.

Reply:
Oh do we also need to parse and set [iam_assume_role_duration](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#iam_assume_role_duration) here?